pkg/apis/serving/v1alpha1/route_defaults_test.go

```diff
@@ -51,13 +54,54 @@ func TestRouteDefaulting(t *testing.T) {
 		want: &Route{
 			Spec: RouteSpec{
 				Traffic: []TrafficTarget{{
+					DeprecatedName: "foo",
 					TrafficTarget: v1beta1.TrafficTarget{
 						ConfigurationName: "foo",
 						Percent:           50,
 						LatestRevision:    ptr.Bool(true),
 					},
 				}, {
 					TrafficTarget: v1beta1.TrafficTarget{
+						Tag:            "bar",
+						RevisionName:   "foo",
+						Percent:        50,
+						LatestRevision: ptr.Bool(false),
+					},
+				}},
+			},
+		},
+	}, {
+		name: "lemonade",
+		wc:   v1beta1.WithUpgradeViaDefaulting,
+		in: &Route{
+			Spec: RouteSpec{
+				Traffic: []TrafficTarget{{
+					DeprecatedName: "foo",
```

Comment:
Can you add a test where this is "bar"? IIRC this should result in two entries that will then error in our validation, correct? Just want to make sure we don't end up with a single entry.

Reply:
Added that case and one other.